packages/jellyfish-api-core/__tests__/category/loan/setDefaultLoanScheme.test.ts

```diff
@@ -0,0 +1,115 @@
+    await testing.container.stop()
+  })
+
+  it('should setDefaultLoanScheme', async () => {
+    const loanSchemeId = await testing.rpc.loan.setDefaultLoanScheme('scheme2')
+    expect(typeof loanSchemeId).toStrictEqual('string')
+    expect(loanSchemeId.length).toStrictEqual(64)
+    await testing.generate(1)
+
+    const result = await testing.container.call('listloanschemes', [])
+    expect(result).toStrictEqual(
+      [
+        { id: 'scheme1', mincolratio: 100, interestrate: 1.5, default: false },
+        { id: 'scheme2', mincolratio: 200, interestrate: 2.5, default: true }
+      ]
+    )
+  })
+
+  it('should not setDefaultLoanScheme if id is an empty string', async () => {
+    const promise = testing.rpc.loan.setDefaultLoanScheme('')
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test DefaultLoanSchemeTx execution failed:\nid cannot be empty or more than 8 chars long\', code: -32600, method: setdefaultloanscheme')
+  })
+
+  it('should not setDefaultLoanScheme if id is more than 8 chars long', async () => {
+    const promise = testing.rpc.loan.setDefaultLoanScheme('123456789')
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test DefaultLoanSchemeTx execution failed:\nid cannot be empty or more than 8 chars long\', code: -32600, method: setdefaultloanscheme')
+  })
+
+  it('should not setDefaultLoanScheme if id does not exists', async () => {
+    const promise = testing.rpc.loan.setDefaultLoanScheme('scheme3')
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test DefaultLoanSchemeTx execution failed:\nCannot find existing loan scheme with id scheme3\', code: -32600, method: setdefaultloanscheme')
+  })
+
+  it('should not setDefaultLoanScheme if the scheme is a default scheme', async () => {
+    const promise = testing.rpc.loan.setDefaultLoanScheme('scheme1')
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test DefaultLoanSchemeTx execution failed:\nLoan scheme with id scheme1 is already set as default\', code: -32600, method: setdefaultloanscheme')
+  })
+
+  it('should not setDefaultLoanScheme if the scheme is going to be deleted at future block', async () => {
+    // Wait for block 110
+    await testing.container.waitForBlockHeight(110)
+
+    await testing.container.call('createloanscheme', [300, new BigNumber(3.5), 'scheme3'])
+    await testing.generate(1)
+
+    // To delete at block 120
+    const loanSchemeId = await testing.container.call('destroyloanscheme', ['scheme3', 120])
+    expect(typeof loanSchemeId).toStrictEqual('string')
+    expect(loanSchemeId.length).toStrictEqual(64)
+    await testing.generate(1)
+
+    const promise = testing.rpc.loan.setDefaultLoanScheme('scheme3')
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test DefaultLoanSchemeTx execution failed:\nCannot set scheme3 as default, set to destroyed on block 120\', code: -32600, method: setdefaultloanscheme')
+
+    // Delete at block 120
+    await testing.container.waitForBlockHeight(120)
```

Comment:
redundant.

Reply:
If I don't do so, the code written in this `it` maybe is going to be executed in the next `it` depending on how many block generated. It is a clean up code, to make sure everything set in this `it` executed here

Reply:
I see. ideally, tests(`it()`) should not affect each other. but since we are using the same container, this might be hard to achieve fully. but we try to minimize the influence on each other and to write more independent tests. 
In your case, the cleanup code should be inside an `afterEach` block. but here the cleanup depends on the height. not sure if we can do something like this https://github.com/DeFiCh/jellyfish/blob/a980b3b2aa15c6f13f8380aa366333b5d11b1133/packages/jellyfish-api-core/__tests__/category/icxorderbook/icx_setup.ts#L119

If not, you can keep it since changing the test execution order does not affect other tests.
or else you can change the comparison logic(`expects`) in other tests.

Reply:
Ya Let's see what Fuxing think

Reply:
I will apply the same changes to all my rpcs and dftxs

Reply:
I agree with @surangap best case scenario is always separating them, I would just create another container for this test. Writing it like this works for now but is difficult to maintain in the long run. Culturally if we take an extreme stance of writing independent tests it makes it an authoritative norm that all tests must be independent. A win-win-win for everyone that maintains this codebase.

But I am confused about the utility of destroying a loan scheme, wondering what purpose does it serve. Can you even destroy a loan scheme with an active loan?

Reply:
Found the reasoning in https://github.com/DeFiCh/pinkpaper/tree/main/loan

I will merge these, for @surangap point should tackle in your next PRs

Reply:
Key sentence: Any loans under the scheme will be moved to default loan scheme after destruction.